chromium_src/components/search_engines/template_url_prepopulate_data.cc

```diff
@@ -53,61 +87,204 @@ const std::map<BravePrepopulatedEngineID, const PrepopulatedEngine*>
+        {country_codes::CountryCharsToCountryID('G', 'B'),
+         brave_engines_with_yahoo},
+        {country_codes::CountryCharsToCountryID('H', 'K'),
+         brave_engines_with_yahoo},
+        {country_codes::CountryCharsToCountryID('I', 'D'),
+         brave_engines_with_yahoo},
+        {country_codes::CountryCharsToCountryID('I', 'E'),
+         brave_engines_AU_NZ_IE},
+        {country_codes::CountryCharsToCountryID('I', 'N'),
+         brave_engines_with_yahoo},
+        {country_codes::CountryCharsToCountryID('I', 'T'),
+         brave_engines_with_yahoo},
+        {country_codes::CountryCharsToCountryID('M', 'X'),
+         brave_engines_with_yahoo},
+        {country_codes::CountryCharsToCountryID('M', 'Y'),
+         brave_engines_with_yahoo},
+        {country_codes::CountryCharsToCountryID('N', 'L'),
+         brave_engines_with_yahoo},
+        {country_codes::CountryCharsToCountryID('N', 'O'),
+         brave_engines_with_yahoo},
         {country_codes::CountryCharsToCountryID('N', 'Z'),
          brave_engines_AU_NZ_IE},
+        {country_codes::CountryCharsToCountryID('P', 'E'),
+         brave_engines_with_yahoo},
+        {country_codes::CountryCharsToCountryID('P', 'H'),
+         brave_engines_with_yahoo},
+        {country_codes::CountryCharsToCountryID('S', 'E'),
+         brave_engines_with_yahoo},
+        {country_codes::CountryCharsToCountryID('S', 'G'),
+         brave_engines_with_yahoo},
+        {country_codes::CountryCharsToCountryID('T', 'H'),
+         brave_engines_with_yahoo},
+        {country_codes::CountryCharsToCountryID('T', 'W'),
+         brave_engines_with_yahoo},
+        {country_codes::CountryCharsToCountryID('U', 'S'),
+         brave_engines_with_yahoo},
+        {country_codes::CountryCharsToCountryID('V', 'E'),
+         brave_engines_with_yahoo},
+        {country_codes::CountryCharsToCountryID('V', 'N'),
+         brave_engines_with_yahoo}};
+
+// Default search engine. Overridable on a per-country basis.
+const BravePrepopulatedEngineID default_engine =
+    PREPOPULATED_ENGINE_ID_GOOGLE;
+
+// A map tracking the singular default search engine per-country.
+const std::map<int, BravePrepopulatedEngineID>
+    default_engine_by_country_id_map = {
+        {country_codes::CountryCharsToCountryID('A', 'U'),
+         PREPOPULATED_ENGINE_ID_DUCKDUCKGO_AU_NZ_IE},
+        {country_codes::CountryCharsToCountryID('F', 'R'),
+         PREPOPULATED_ENGINE_ID_QWANT},
+        {country_codes::CountryCharsToCountryID('D', 'E'),
+         PREPOPULATED_ENGINE_ID_DUCKDUCKGO_DE},
         {country_codes::CountryCharsToCountryID('I', 'E'),
-         brave_engines_AU_NZ_IE}};
+         PREPOPULATED_ENGINE_ID_DUCKDUCKGO_AU_NZ_IE},
+        {country_codes::CountryCharsToCountryID('N', 'Z'),
+         PREPOPULATED_ENGINE_ID_DUCKDUCKGO_AU_NZ_IE}};
+
+// A map to keep track of country-specific implementations of Yahoo.
+// Used in LocalizeEngineList.
+const std::map<int, BravePrepopulatedEngineID>
```

Comment:
There is a bunch of global non-trivial objects already existing in this file, but can we please add new ones properly - i.e. wrapped into a function and `base::NoDestructor`? Also, constants should be named as `kYahoo...` according to CC, this really prevents confusion.

Reply:
also `flat_map` should be preferred to `std::map`

Reply:
Thanks for this feedback! I had it bookmarked and am checking out some search changes 😄 

It's still WIP (so don't be harsh!) but see https://github.com/brave/brave-core-private/pull/9 and `GetLocalizedEngineYahoo` or `GetDefaultSearchEngine` for an example. Guessing this is more like what you had in mind?

Reply:
@bsclifton yes, absolutely!